lib/errors/warning.go

```diff
@@ -0,0 +1,35 @@
+package errors
+
+// Warning embeds an error. It's purpose is to indicate that this error is not a critical error and
+// maybe ignored. Additionally, it **must** be logged only as a warning. If it cannot be logged as a
+// warning, then these are not the droids you're looking for.
+type Warning interface {
+	error
+	IsWarn() bool
+}
+
+type warning struct {
+	Err error
+}
+
+// Ensure that warning always implements the error interface.
+var _ error = (*warning)(nil)
+
+// Ensure that warning always implements the Warning interface.
+var _ Warning = (*warning)(nil)
```

Comment:
`Warning` already embeds the `error` interface so I think we can remove the first assertion
```suggestion
// Ensure that warning always implements the Warning error interface.
var _ Warning = (*warning)(nil)
```